### PR TITLE
Add simple unit tests

### PR DESCRIPTION
**Pull request by BShaq**
I have added some simple unit tests that will increase code coverage / branch coverage.
They contribute to issue #6207

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [X] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java [hunk 1]
```diff
@@ -173,4 +173,97 @@ public void compareStringFieldsBiggerDescending() throws Exception {
 
         assertEquals(1, comparator.compare(bigger, smaller));
     }
+
+    @Test
+    public void compareNumericFieldsIdentity() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.PMID);
+        BibEntry equal = new BibEntry();
+        equal.setField(StandardField.PMID, "123456");
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java:181]: BibEntry equal = new BibEntry().withField(...)
```suggestion
        BibEntry equal = new BibEntry()
        	.withField(StandardField.PMID, "123456");
```

Comment (BShaq) [reply src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java]: Hi @calixtus 
Thank you for reviewing and for this suggestion!

I've applied it to the rest of the code (commit 25c47df).

File: src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java [hunk 2]
```diff
@@ -135,42 +135,134 @@ public void compareTypeFieldBiggerDescending() throws Exception {
...
+
+    @Test
+    public void compareNumericFieldsEquality() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.PMID);
+        BibEntry equal = new BibEntry()
+                .withField(StandardField.PMID, "123456");
+        BibEntry equal2 = new BibEntry()
+                .withField(StandardField.PMID, "123456");
+
+        assertEquals(0, comparator.compare(equal, equal2));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java:194]: Nearly all tests have the following form:

```suggestion
        FieldComparator comparator = new FieldComparator(StandardField.PMID);
        BibEntry ENTRY1 = new BibEntry()
                .withField(StandardField.PMID, ID1);
        BibEntry ENTRY2 = new BibEntry()
                .withField(StandardField.PMID, ID2);

        assertEquals(INT, comparator.compare(ENTRY1, ENTRY2));
```

Please convert these to a parameterized tests with INT, ID1, ID2 as input. (replace by `comparisonResult, id1, id2` in the implementation.

Hint:

```java
Stream.of(
  Arguments.of(0, "123456", "123456")
  ...
```

Full example at https://stackoverflow.com/a/61484155 (look at the second one 🌞 )

File: src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeAmpersandsFormatterTest.java [hunk 3]
```diff
@@ -28,4 +28,14 @@ void formatEscapesAmpersandsIfPresent() throws Exception {
...
         assertEquals("Text \\& with \\&ampersands", formatter.format(formatter.getExampleInput()));
     }
+
+    @Test
+    void formatReturnsSameTextInNewUserDefinedLatexCommandIfNoAmpersandsPresent() throws Exception {
+        assertEquals("\\newcommand[1]{Lorem ipsum}", formatter.format("\\newcommand[1]{Lorem ipsum}"));
+    }
+
+    @Test
+    void formatReturnsSameTextInLatexCommandIfPresent() throws Exception {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeAmpersandsFormatterTest.java:38]:
```suggestion
    void formatReturnsSameTextInLatexCommandIfOneAmpersandPresent() throws Exception {
```

File: src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java [hunk 4]
```diff
@@ -19,7 +19,7 @@ public void testNull() {
 
     @Test
     public void testLowerCase() {
-        assertEquals("abcd efg", new ToLowerCase().format("abcd efg"));
+        assertEquals("abcd efg ", new ToLowerCase().format("abcd efg "));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java:22]: Please do not modify existing test cases. Please revert this change.

If you think, this is important, please add another test case.

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you for the work; some small comments remain ^^

File: src/test/java/org/jabref/model/entry/DateTest.java [hunk 5]
```diff
@@ -40,4 +40,29 @@ void parseCorrectlyYearDate() throws Exception {
     void parseDateNull() {
         assertThrows(NullPointerException.class, () -> Date.parse(null));
     }
+
+    @Test
+    void emptyInputValue() {
```
Comment (koppor) [inline src/test/java/org/jabref/model/entry/DateTest.java:45]: Please use parameterized tests, here, too.

File: src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java [hunk 6]
```diff
@@ -31,4 +31,19 @@ public void testUpperCase() {
     public void testMixedCase() {
         assertEquals("abcd efg", new ToLowerCase().format("abCD eFg"));
     }
+
+    @Test
+    public void includeNumbersInString() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java:36]: Please use parameterized tests, here, too.

File: src/test/java/org/jabref/logic/layout/format/ShortMonthFormatterTest.java [hunk 7]
```diff
@@ -17,8 +17,37 @@ public void setUp() {
     }
 
     @Test
-    public void testFormat() {
-        assertEquals("jan", formatter.format("01"));
+    public void formatShortName() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/ShortMonthFormatterTest.java:20]: Please use parameterized tests, here, too.

File: src/test/java/org/jabref/logic/integrity/ISSNCheckerTest.java [hunk 8]
```diff
@@ -31,4 +33,24 @@ void issnDoesNotAcceptInvalidInput() {
         assertNotEquals(Optional.empty(), checker.checkValue("0020-7218"));
     }
 
+    @Test
+    void lessThanEightDigitCode() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/integrity/ISSNCheckerTest.java:37]: Please use parameterized tests, here, too.

File: src/test/java/org/jabref/logic/integrity/ISBNCheckerTest.java [hunk 9]
```diff
@@ -31,4 +33,18 @@ void isbnDoesNotAcceptInvalidInput() {
         assertNotEquals(Optional.empty(), checker.checkValue("0-201-53082-2"));
     }
 
+    @Test
```
Comment (koppor) [inline src/test/java/org/jabref/logic/integrity/ISBNCheckerTest.java:36]: Please use parameterized tests, here, too. 😇

**Comment by BShaq** [comment]
Hi @koppor,
thanks for reviewing!
I've addressed your comments in commit 9492146 and 51b7d8d.

File: src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java (hunk 4 above)
Comment (BShaq) [reply src/test/java/org/jabref/logic/layout/format/ToLowerCaseTest.java]: I've included it in the parameterized test now, is that OK or shall I revert it to its original form?

**Review by koppor — APPROVED** [review]
